src/java.desktop/windows/native/libawt/windows/awt_Frame.cpp

```diff
@@ -1567,26 +1578,63 @@ void AwtFrame::_SetIMMOption(void *param)
 
 void AwtFrame::_NotifyModalBlocked(void *param)
 {
-    JNIEnv *env = (JNIEnv *)JNU_GetEnv(jvm, JNI_VERSION_1_2);
+    JNIEnv *env = (JNIEnv *) JNU_GetEnv(jvm, JNI_VERSION_1_2);
 
-    NotifyModalBlockedStruct *nmbs = (NotifyModalBlockedStruct *)param;
+    NotifyModalBlockedStruct *nmbs = static_cast<NotifyModalBlockedStruct *>(param);
     jobject self = nmbs->frame;
     jobject peer = nmbs->peer;
     jobject blockerPeer = nmbs->blockerPeer;
     jboolean blocked = nmbs->blocked;
 
     PDATA pData;
 
-    JNI_CHECK_PEER_GOTO(peer, ret);
-    AwtFrame *f = (AwtFrame *)pData;
+    if (peer == NULL) {
+        env->ExceptionClear();
+        JNU_ThrowNullPointerException(env, "peer");
+        env->DeleteGlobalRef(self);
+        env->DeleteGlobalRef(blockerPeer);
+
+        delete nmbs;
+        return;
+    } else {
+        pData = JNI_GET_PDATA(peer);
```

Comment:
like in my recoded case above, we no longer need the "pData" var which was there only because that name
is magically known to the macro.